docs/backend/api-definitions-guidelines.md

```diff
@@ -0,0 +1,58 @@
+# API Definitions Guidelines
+
+This document describes how service APIs will be managed for `InstructLab` and the sub-components of the `InstructLab` backend.
+
+## What parts of InstructLab need service APIs?
+
+There are two primary classes of service APIs needed to support InstructLab:
+
+* `Platform APIs`: These are APIs that are ignorant of `InstructLab` and provide generic AI platform capabilities (e.g., Fine Tuning, SDG, Eval)
+* `InstructLab APIs`: These are the APIs that reflect the user-facing functionality of `InstructLab` itself. They are aware of the end-to-end `InstructLab` workflow.
+
+The `InstructLab APIs` are essential for hosting `InstructLab` as a service in a repeatable way. The `Platform APIs` are critical for component reuse and extensibility (e.g., new SDG algorithms for new taxonomy data types), but they are not strictly required for hosting `InstructLab` as a service.
+
+## How will service APIs be defined?
+
+Service APIs will be defined using [OpenAPI](https://www.openapis.org/) format in [YAML](https://yaml.org/). For structural and style guidelines, see [api-definitions](../../api-definitions/README.md).
+
+## Where will service API definitions live?
+
+Service API definitions will live a new repository github.com/instructlab/service-api-definitions. This repo will have two primary responsibilities:
+
+1. House the static service API definitions
+2. Build and publish any language-specific generated packages for consumption by service implementation projects (see below)
```

Comment:
When I read "build and publish," I understand it to mean that the generated files will be hosted in the service-api-definitions repository. I’m not sure this is the best approach.

I can see two scenarios:

The current approach, where the generated code is hosted in the service-api-definitions repository.
An alternative where we still build and publish, but to a dedicated repository. This option provides a clear separation of concerns and prevents the API spec repository from becoming bloated.

Did I understand correctly? If so, my preference would be option 2. Thanks!

Reply:
Probably just a poor choice of words. It's probably language specific, and I'm less familiar with modern `go` package management (when I was writing it a number of years ago, it was still "grab the latest from the tip of master in the repo). For `python`, I was imagining the CI generating the python stubs (build) and putting them in a package that gets published to PyPI (publish).